app/generic/definitions.py

```diff
@@ -0,0 +1,390 @@
+import asyncio
+import json
+from typing import List, Optional
+
+from aries_cloudcontroller import (
+    AcaPyClient,
+    CredentialDefinition as AcaPyCredentialDefinition,
+    ModelSchema,
+    SchemaSendRequest,
+    SchemaSendResult,
+    TxnOrCredentialDefinitionSendResult,
+)
+from app.error.cloud_api_error import CloudApiException
+from aries_cloudcontroller.model.credential_definition_send_request import (
+    CredentialDefinitionSendRequest,
+)
+from fastapi import APIRouter, Depends, HTTPException
+from pydantic import BaseModel, Field
+
+from app.dependencies import (
+    AcaPyAuthVerified,
+    acapy_auth_verified,
+    agent_role,
+    agent_selector,
+)
+from app.role import Role
+from app.facades import trust_registry, acapy_wallet
+from app.webhook_listener import start_listener
+
+router = APIRouter(
+    prefix="/generic/definitions",
+    tags=["definitions"],
+)
+
+
+class CreateCredentialDefinition(BaseModel):
+    # Revocation not supported currently
+    # support_revocation: bool = False
+    tag: str = Field(..., example="default")
+    schema_id: str = Field(..., example="CXQseFxV34pcb8vf32XhEa:2:test_schema:0.3")
+
+
+class CredentialDefinition(BaseModel):
+    id: str = Field(..., example="5Q1Zz9foMeAA8Q7mrmzCfZ:3:CL:7:default")
+    tag: str = Field(..., example="default")
+    schema_id: str = Field(..., example="CXQseFxV34pcb8vf32XhEa:2:test_schema:0.3")
+
+
+class CreateSchema(BaseModel):
+    name: str = Field(..., example="test_schema")
+    version: str = Field(..., example="0.3.0")
+    attribute_names: List[str] = Field(..., example=["speed"])
+
+
+class CredentialSchema(BaseModel):
+    id: str = Field(..., example="CXQseFxV34pcb8vf32XhEa:2:test_schema:0.3")
+    name: str = Field(..., example="test_schema")
+    version: str = Field(..., example="0.3.0")
+    attribute_names: List[str] = Field(..., example=["speed"])
+
+
+def _credential_schema_from_acapy(schema: ModelSchema):
+    return CredentialSchema(
+        id=schema.id,
+        name=schema.name,
+        version=schema.version,
+        attribute_names=schema.attr_names,
+    )
+
+
+def _credential_definition_from_acapy(credential_definition: AcaPyCredentialDefinition):
+    return CredentialDefinition(
```

Comment:
or as below comment (maybe good to keep this method as you're reusing it
```python
def _credential_definition_from_acapy(credential_definition: AcaPyCredentialDefinition):
    return CredentialDefinition(**credential_definition)
```

Reply:
I'm not really a fan of this implicit approach tbh. We're transforming properties between two different data models, and although the properties currently match they may drift apart over time. It also explains to the developer how we're transforming from the acapy model to our own model. So I mainly do it to be explicit.